subworkflows/nf-core/bam_baserecalibrator/tests/main.nf.test

```diff
@@ -0,0 +1,44 @@
+nextflow_workflow {
+
+    name "Test Subworkflow BAM_BASERECALIBRATOR"
+    script "../main.nf"
+    workflow "BAM_BASERECALIBRATOR"
+
+    tag "subworkflows"
+    tag "subworkflows_nfcore"
+    tag "subworkflows/bam_baserecalibrator"
+
+    tag "gatk4/baserecalibrator"
+    tag "gatk4/gatherbqsrreports"
+
+    test("sarscov2 - bam - paired_end") {
+
+        when {
+            workflow {
+                """
+                input[0] = Channel.of([
+                    [ id:'test' ], // meta map
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true),
+                ])
+                input[1] = Channel.of(file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.dict', checkIfExists: true))
+                input[2] = Channel.of(file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true))
+                input[3] = Channel.of(file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta.fai', checkIfExists: true))
+                input[4] = Channel.of([
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true),
+                    4
+                    ])
```

Comment:
Given the sneaky behavior of merging and not merging, I would vote for having a few more tests here: 

- 1 interval
- 0 as number
- what happens if the intervals are left empty? They are technically not necessary for bqsr afaik

Reply:
I added empty intervals with 0 and 1 as interval. Maybe you can check if that is what you meant.